core/services/pipeline/runner.go

```diff
@@ -367,26 +404,28 @@ func (r *runner) executeTaskRun(ctx context.Context, txdb *gorm.DB, taskRun Task
 		return Result{Error: err}
 	}
 
-	spec := taskRun.PipelineTaskSpec.PipelineSpec
-
 	// Order of precedence for task timeout:
 	// - Specific task timeout (task.TaskTimeout)
 	// - Job level task timeout (spec.MaxTaskDuration)
 	// - Passed in context
 	taskTimeout, isSet := task.TaskTimeout()
+	specificTaskCtx := context.Background()
 	if isSet {
 		var cancel context.CancelFunc
-		ctx, cancel = utils.CombinedContext(r.chStop, taskTimeout)
+		specificTaskCtx, cancel = utils.CombinedContext(r.chStop, taskTimeout)
 		defer cancel()
 	} else if spec.MaxTaskDuration != models.Interval(time.Duration(0)) {
 		var cancel context.CancelFunc
-		ctx, cancel = utils.CombinedContext(r.chStop, time.Duration(spec.MaxTaskDuration))
+		specificTaskCtx, cancel = utils.CombinedContext(r.chStop, time.Duration(spec.MaxTaskDuration))
 		defer cancel()
 	}
+	ctx, cancel := utils.CombinedContext(ctx, specificTaskCtx)
+	defer cancel()
```

Comment:
@connorwstein I cannot remember, is this the correct fix?

Reply:
nah - ill push up a commit